### PR TITLE
[POAE7-2455] Add a tool to dump substrait plan, llvm ir and llvm cfg

### PR DESCRIPTION
**Pull request by Ma-Jian1**
### What changes were proposed in this pull request?
add a tool to dump substrait plan, llvm ir and llvm cfg

### Why are the changes needed?
the tool can help to understand code

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
UTs

### Which label does this PR belong to?
INFRA

### CODE REVIEW

File: cider/exec/template/CompilationOptions.h [hunk 1]
```diff
@@ -62,7 +62,7 @@ enum class ExecutorType { Native, Extern, TableFunctions };
 struct ExecutionOptions {
   bool output_columnar_hint;
   bool allow_multifrag;
-  bool just_explain;  // return the generated IR for the first step
+  uint32_t just_explain;  // return the generated IR for the first step
```
Comment (bigPYJ1151) [inline cider/exec/template/CompilationOptions.h:65]: Maybe ```enum``` is more explicit. Just an advice.

**Review by bigPYJ1151 — APPROVED** [review]

File: cider/exec/template/CompilationOptions.h (hunk 1 above)
Comment (Ma-Jian1) [reply cider/exec/template/CompilationOptions.h:65]: yep

File: cider/exec/template/NativeCodegen.cpp [hunk 2]
```diff
@@ -1722,27 +1715,33 @@ Executor::compileWorkUnit(const std::vector<InputTableInfo>& query_infos,
...
 #endif
+    } else if (eo.just_explain == EXPLAIN_MODULE) {
+      llvm_ir = serialize_llvm_object(cgen_state_->module_);
+    }
   }
 
   LOG(IR) << "\n\n" << query_mem_desc->toString() << "\n";
-  LOG(DEBUG1) << "IR for the CPU: \n";
 #ifndef NDEBUG
+  LOG(DEBUG1) << "IR for the CPU: \n";
```
Comment (winningsix) [inline cider/exec/template/NativeCodegen.cpp:1736]: Just remove this since we're only CPU?

File: cider/tests/Sql2IR.cpp [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+    col_names.emplace_back(schema.names(i));
+    col_types.emplace_back(schema.struct_().types(i));
+  }
+
+  auto table_schema = std::make_shared<CiderTableSchema>(col_names, col_types);
+
+  cider_compile_module->feedBuildTable(CiderBatch(1, table_schema));
+}
+
+int main(int argc, char** argv) {
```
Comment (winningsix) [inline cider/tests/Sql2IR.cpp:93]: Better to add a tutorial in https://intel.github.io/BDTK/develop.html and make other aware of this great tool.

**Review by winningsix — APPROVED** [review]

File: cider/exec/template/NativeCodegen.cpp (hunk 2 above)
Comment (Ma-Jian1) [reply cider/exec/template/NativeCodegen.cpp:1736]: I think it's also used for the header of IR in the log

**Review by winningsix — APPROVED** [review]

**Review by bigPYJ1151 — APPROVED** [review]

**Pull request merged**